README.md

```diff
@@ -1,2 +1,164 @@
+
+```HTML
+<dfn id="simple-def" data-export="" data-dfn-type="dfn">
+  I'm a simple definition
+</dfn>
+```
+
+A more complex examples:
+
+```HTML
+<!-- Example of defining an abstract operation -->
+<dfn
+  id="string-parser"
+  data-export=""
+  data-lt="parse a string"
+  data-dfn-type="abstract-op">
+  string parser
+</dfn>
+
+<!-- Example of defining a WebIDL interface -->
+<dfn
+  data-export=""
+  data-dfn-type="interface"
+  id="dom-paymentrequest"
+  data-dfn-for="">
+  <code>PaymentRequest</code>
+</dfn>
+
+```
+
+## Conformance
+
+This document serves as the canonical source for how definitions are marked-up.
+
+Conforming applications to this specification are primarily authoring tools that assist in generating technical specifications. However, this specification is also here to help people who wish to mark up documents manually (or simply want to understand why various attributes mean and what they do!).
+
+Example of authoring tools (or "generators") that try to conform to this specification:
+
+- [Bikeshed](http://github.com/tabatkins/bikeshed)
+- [ReSpec](http://github.com/w3c/respec)
+- [Wattsi](https://github.com/whatwg/wattsi)
+
+Note: An accompanying test suite helps assure authoring tools conform to the contract.
+
+## Who consumes these definitions
+
+Aside from specifications internally cross-referencing their own terms and terms in other specifications, "crawlers" extract terms defined in specifications into structured data.
+
+Examples of definition crawlers include:
+
+- [Shepherd](https://dev.csswg.org/projects/shepherd)
+- [Reffy](https://github.com/w3c/reffy)
+
+Crawlers structure and categorized extracted terms into files or databases, allowing for the creation of search tools such as:
+
+- <https://respec.org/xref>
+- Bikeshed's cross-reference search.
+
+## How to get your spec indexed
+
+To have your specification indexed by a crawler, it MUST be registered with:
+
+- [browser specs](https://github.com/w3c/browser-specs/) - See [criteria for inclusion](https://github.com/w3c/browser-specs/blob/master/README.md#spec-selection-criteria).
+- [Shepherd](https://dev.csswg.org/projects/shepherd) - [Contact maintainer(s) directly](https://dev.csswg.org/users/3).
+
+## The Contract
+
+The following sections define the elements, attributes, and attribute values (along with any specific micro-syntax) that constitute the contract.
+
+### Definitions
+
+A <dfn>definition</dfn> is:
+
+- a `<dfn>` element.
+- a `h1`-`h6` element.
+
+No other HTML elements are recognized as defining a term.
+
+#### dfn
+A a dfn-definition MUST have the following attributes:
+
+- `id` attribute, unique to the document.
+- `data-dfn-type`, with one of type values.
+
+The following attributes are OPTIONAL:
+
+- `data-export` or `data-noexport` (they are exclusive).
+- `data-dfn-for`
+
+### Exporting definitions (`data-export`)
+
+Exported definitions MUST include a `data-export` attribute. The attribute's value can be missing or the empty string.
```

Comment:
this should recognize the intersection with `dfn-type` that makes some definitions exported without a `data-export` attribute

Reply:
Hmm... that seems to be a Bikeshed-ism that relies on tree structure (ReSpec doesn't support `dfn-type` on container elements). 

I think the tooling should be adding `data-export` in those situations.

Reply:
I agree with @marcoscaceres and think that's already the case in practice, `dfn-type` being just a tooling artefact to produce `data-dfn-type` attributes in the resulting spec. Typically, Reffy does not know anything about `dfn-type` and only looks for `data-dfn-type` attributes. I don't think that the contract document should mention `dfn-type` at all.

Reply:
let me clarify what I meant - I wasn't arguing about `dfn-type` vs `data-dfn-type` (I was just foolishly saving typing 5 characters).  I was saying some value of `data-dfn-type` are exported by default, some are not exported by default, and this should be reflected in the definition of exported definitions.

Reply:
Got it, thanks for clarifying @dontcallmedom. 

So, let's reframe this because I think this is really important. Tooling can add `data-dfn-type`, but (!!!) they must add `data-export=""` for a definition to actually be classified as exported. 

Put differently, if a definition is "exported" must not be implied by the presence of `data-dfn-type`. Only the explicit presence of `data-export=""` makes something exported.  

Agree? 

I know that adds redundancy, but it also gets rid of any ambiguity.

Reply:
Ah, I understand what you're suggesting now: the fact that some `data-dfn-type` are automatically exported is not a rule of the type, but a default that authoring tools apply when generating the output.

This has the merits of disentangling the two axes; I'm a bit worried of the impact on making this work for non-bikeshed / non-respec maintained documents, but we can revisit this when this becomes a more realistic question to consider :)

Reply:
Yeah, putting the dfn-* attributes on ancestors is a Bikeshed-ism; it gets turned into `data-dfn-*` attributes directly on the `<dfn>` by generation time.

A *mention* of this is very likely reasonable, in case people run into this document when looking for how to format their dfns, but it's definitely not part of the contract itself.

> Put differently, if a definition is "exported" must not be implied by the presence of data-dfn-type. Only the explicit presence of data-export="" makes something exported.

I dunno if I agree with this. I think in the absence of an explicit export/noexport, it's reasonable to apply the same defaulting that Bikeshed does. (But we should still require tooling to add it explicitly.)